psl-parser/src/main/java/org/linqs/psl/parser/CommandLineLoader.java

```diff
@@ -0,0 +1,421 @@
+                    name1 = o1.getLongOpt();
+                }
+
+                String name2 = o2.getOpt();
+                if (name2 == null) {
+                    name2 = o2.getLongOpt();
+                }
+
+                if (name1.equals(OPERATION_INFER)) {
+                    return -1;
+                }
+
+                if (name2.equals(OPERATION_INFER)) {
+                    return 1;
+                }
+
+                if (name1.equals(OPERATION_LEARN)) {
+                    return -1;
+                }
+
+                if (name2.equals(OPERATION_LEARN)) {
+                    return 1;
+                }
+
+                if (o1.isRequired() && !o2.isRequired()) {
+                    return -1;
+                }
+
+                if (!o1.isRequired() && o2.isRequired()) {
+                    return 1;
+                }
+
+                return name1.compareTo(name2);
+            }
+        });
+
+        helpFormatter.setWidth(100);
+
+        return helpFormatter;
+    }
+
+    /**
+     * Parse the options on the command line.
+     * Will exit on error, but Will return null if the CLI should not be run (like if we are doing a help/version run).
+     */
+    private static CommandLine parseOptions(String[] args) {
+        Options options = setupOptions();
+        CommandLineParser parser = new DefaultParser();
+        CommandLine commandLineOptions = null;
+
+        try {
+            commandLineOptions = parser.parse(options, args);
+        } catch (ParseException ex) {
+            System.err.println("Command line error: " + ex.getMessage());
+            getHelpFormatter().printHelp("psl", options, true);
+            System.exit(1);
+        }
+        if (commandLineOptions.hasOption(OPTION_HELP)) {
+            initDefaultLogger();
+            getHelpFormatter().printHelp("psl", options, true);
+            return commandLineOptions;
+        }
+        if (commandLineOptions.hasOption(OPTION_VERSION)) {
+            initDefaultLogger();
+            System.out.println("PSL CLI Version " + Version.getFull());
+            return commandLineOptions;
+        }
+        // Can't have both an H2 and Postgres database.
+        if (commandLineOptions.hasOption(OPTION_DB_H2_PATH) && commandLineOptions.hasOption(OPTION_DB_POSTGRESQL_NAME)) {
+            System.err.println("Command line error: Options '--" + OPTION_DB_H2_PATH + "' and '--" + OPTION_DB_POSTGRESQL_NAME + "' are not compatible.");
+            getHelpFormatter().printHelp("psl", options, true);
+            System.exit(2);
+        }
+        return commandLineOptions;
+    }
+
+    private static String getHostname() {
+        String hostname = "unknown";
+
+        try {
+            hostname = InetAddress.getLocalHost().getHostName();
+        } catch (UnknownHostException ex) {
+            // log.warn("Hostname can not be resolved, using '" + hostname + "'.");
+        }
+
+        return hostname;
+    }
+}
```

Comment:
Missing a newline at the end of the file.

Reply:
done

Reply:
Make sure not to reply to comments you don't need to (I get emails for replies).
Just mark them as resolved (all comments should be resolved or commented on).